packages/x-charts/src/internals/plugins/corePlugins/useChartDimensions/useChartDimensions.ts

```diff
@@ -35,53 +48,106 @@ export const useChartDimensions: ChartPlugin<UseChartDimensionsSignature> = ({
     const newWidth = Math.floor(parseFloat(computedStyle.width)) || 0;
 
     store.update((prev) => {
-      const prevWidth = prev.dimensions.width + prev.dimensions.left + prev.dimensions.right;
-      const prevHeight = prev.dimensions.height + prev.dimensions.top + prev.dimensions.bottom;
+      const prevWidth =
+        params.width ?? prev.dimensions.width + prev.dimensions.left + prev.dimensions.right;
+      const prevHeight =
+        params.height ?? prev.dimensions.height + prev.dimensions.top + prev.dimensions.bottom;
 
       if (prevWidth === newWidth && prevHeight === newHeight) {
         return prev;
       }
 
+      const axisSize = defaultizeMargin(
+        {
+          top: topAxisSize,
+          left: leftAxisSize,
+          bottom: bottomAxisSize,
+          right: rightAxisSize,
+        },
+        EMPTY_SIDES,
+      );
+
       return {
         ...prev,
         dimensions: {
-          ...prev.dimensions,
-          width: newWidth - prev.dimensions.left - prev.dimensions.right,
-          height: newHeight - prev.dimensions.top - prev.dimensions.bottom,
+          top: params.margin.top + axisSize.top,
+          left: params.margin.left + axisSize.left,
+          bottom: params.margin.bottom + axisSize.bottom,
+          right: params.margin.right + axisSize.right,
+          width:
+            newWidth - params.margin.left - params.margin.right - axisSize.left - axisSize.right,
+          height:
+            newHeight - params.margin.top - params.margin.bottom - axisSize.top - axisSize.bottom,
+          propsWidth: params.width,
+          propsHeight: params.height,
         },
       };
     });
     return {
       height: newHeight,
       width: newWidth,
     };
-  }, [store, svgRef]);
+  }, [
+    store,
+    svgRef,
+    params.margin.left,
+    params.margin.right,
+    params.margin.top,
+    params.margin.bottom,
+    params.width,
+    params.height,
+    leftAxisSize,
+    rightAxisSize,
+    topAxisSize,
+    bottomAxisSize,
+  ]);
 
-  store.update((prev) => {
-    if (
-      (params.width === undefined || prev.dimensions.propsWidth === params.width) &&
-      (params.height === undefined || prev.dimensions.propsHeight === params.height)
-    ) {
-      return prev;
-    }
+  React.useEffect(() => {
```

Comment:
This `useEffect` block ensures all properties are updated when the react props update, else they would update only on reload.